components/camel-opentelemetry/src/main/java/org/apache/camel/opentelemetry/OpenTelemetryInstrumentedThreadPoolFactory.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.opentelemetry;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.RejectedExecutionHandler;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+
+import io.opentelemetry.context.Context;
+import org.apache.camel.opentelemetry.internal.CurrentContextScheduledExecutorService;
+import org.apache.camel.spi.ThreadPoolFactory;
+import org.apache.camel.spi.ThreadPoolProfile;
+import org.apache.camel.spi.annotations.JdkService;
+import org.apache.camel.support.DefaultThreadPoolFactory;
+
+@JdkService(ThreadPoolFactory.FACTORY)
+public class OpenTelemetryInstrumentedThreadPoolFactory extends DefaultThreadPoolFactory implements ThreadPoolFactory {
+
+    @Override
+    public ExecutorService newCachedThreadPool(ThreadFactory threadFactory) {
+        return Context.taskWrapping(super.newCachedThreadPool(threadFactory));
+    }
+
+    @Override
+    public ExecutorService newThreadPool(
+            int corePoolSize,
+            int maxPoolSize,
+            long keepAliveTime,
+            TimeUnit timeUnit,
+            int maxQueueSize,
+            boolean allowCoreThreadTimeOut,
+            RejectedExecutionHandler rejectedExecutionHandler,
+            ThreadFactory threadFactory)
+            throws IllegalArgumentException {
+
+        ExecutorService executorService = super.newThreadPool(
+                corePoolSize,
+                maxPoolSize,
+                keepAliveTime,
+                timeUnit,
+                maxQueueSize,
+                allowCoreThreadTimeOut,
+                rejectedExecutionHandler,
+                threadFactory);
+
+        return Context.taskWrapping(executorService);
+    }
+
+    @Override
+    public ScheduledExecutorService newScheduledThreadPool(ThreadPoolProfile profile, ThreadFactory threadFactory) {
+        return new CurrentContextScheduledExecutorService(super.newScheduledThreadPool(profile, threadFactory));
```

Comment:
Can you use 

`return Context.wrap(super.newScheduledThreadPool(profile, threadFactory));`

As opentelemtry has a wrap for ScheduledExecutorService

Reply:
I'll try. There's a PR open on OpenTelemetry to move this implementation there. https://github.com/open-telemetry/opentelemetry-java/pull/6712

Reply:
can we add a TODO in the camel code to refer to this so we wont forget to change it when otel comes with this out of the box.

Reply:
@ammachado okay looks like that otel PR is potentially going to be merged. That would be good to get out of the box there. In the mean time we may need this PR in Camel

Reply:
I will submit another PR once https://github.com/open-telemetry/opentelemetry-java/pull/6712 is merged to remove the duplicated code.

Reply:
Okay thanks